patch/save_patches_old.txt

```diff
@@ -0,0 +1,578 @@
+	set #curbuild, 4 ; we're technically not at build 5 yet
+	iflt #build, 4, .build4_0
+	call PrintBuild4VersionCheck ; old/misc.txt
+	jumptable #result
+	dw .build4_0
+	dw .build4_1
+	dw .build4_2
+	dw .build4_3
+	dw AdjustRhyperiorIndex ; build4_4
+	dw DoNothing ; build4_5
+	
+.build4_0
+	call AdjustShuckleAndEeveelutionIndexes
+.build4_1
+	call ReplaceMilkDrink
+.build4_2
+	call RemovePikabluGuyFlag
+.build4_3
+	call CeruleanCapeYellowForestSpawnFlags
+	; fallthrough
+AdjustRhyperiorIndex:
+	set #r0, .mon_adjust_table
+	jump AdjustPokemonIndexes
+
+.mon_adjust_table
+	; upper bound, offset
+	; terminated by -1
+	dh 0xae, 0 ; MAGNEZONE
+	dh 0xb5, 1 ; BELLOSSOM
+	dh 0xb6, -7 ; RHYPERIOR
+	dh -1
+
+AdjustShuckleAndEeveelutionIndexes:
+	set #r0, .mon_adjust_table
+	call AdjustPokemonIndexes
+
+	seek 0x2a16 ; TODO: replace literal
+	set #1, 210
+	set #2, #zero
+	call FlagAction
+
+	seekfwd 32
+	set #1, 210
+	set #2, #zero
+	jump FlagAction
+
+.mon_adjust_table
+	dh 0xc0, 0 ; UMBREON
+	dh 0xc2, 59 ; LEAFEON/GLACEON
+	dh 0xd4, -2 ; SCIZOR
+	dh 0xe9, -1 ; PORYGON2
+	dh 0xfb, 0 ; CELEBI
+	dh 0xfc, 2 ; SYLVEON
+	dh 0xfd, -20 ; PORYGON_Z
+	dh -1
+
+RemovePikabluGuyFlag:
+	set #1, EVENT_FLAGS_OFFSET_v4
+	call GetGameDataPointer
+	seek #result
+
+	set #1, 2192
+	set #2, .flag_adjust_table
+	jump ShiftFlags
+
+.flag_adjust_table
+	dh 2027, 0
+	dh 2191, -1
+	dh -1
+
+CeruleanCapeYellowForestSpawnFlags:
+	seek sMapData_v4 ; same as Visited Spawns offset
+	ifeq #savefile, 1, .done
+	seek sBackupMapData_v4
+.done
+	set #1, 29
+	set #2, .flag_adjust_table
+	jump ShiftFlags
+
+.flag_adjust_table
+	dh 6, 0
+	dh 24, 1
+	dh 27, 2
+	dh 28, -21
+	dh 29, -3
+	dh -1
+
+RemoveExtraOptionsByte:
+	seek sCheckValue1 ; new check value location
+	set #1, 8183
+	ifeq #savefile, 1, .done
+	seek sBackupCheckValue1
+	set #1, 3575
+.done
+	set #2, 1
+	jump ShiftSaveDataBack
+
+ResetInitialOptions:
+	seek sCheckValue1
+	ifeq #savefile, 1, .done
+	seek sBackupCheckValue1
+.done
+	seekback 1 ; initial options 2
+	writebyte 0x80 ; 0b11111111
+	print .reset_options_string
+	return
+
+.reset_options_string
+	string "*** New initial options were added 1 March 2020. You will therefore be asked to reset your in-game options. ***"
+
+AdjustVariableSprites:
+	set #curbuild, 5
+	set #1, MAP_FROM_CONT_OFFSET_v5
+	call GetGameDataPointer
+	set #3, #result
+	set #1, 7321
+	ifeq #savefile, 1, .done
+	set #1, 2713
+.done
+	set #2, 1
+	call ShiftSaveDataForward
+	seek #result
+	writebyte #zero
+	return
+
+LegendaryDogStatus:
+	set #1, POKEDEX_CAUGHT_OFFSET
+	call GetGameDataPointer
+	seek #result
+
+	set #1, 323
+	set #2, 3
+	call ShiftSaveDataBack
+	writebyte #zero
+
+	set #1, 10
+	decrement #2
+	call ShiftSaveDataBack
+	writebyte #zero
+	
+	set #1, 10
+	decrement #2
+	call ShiftSaveDataBack
+	writebyte #zero
+	return
+
+ChangeTerminatingCharacter:
+	set #1, PLAYER_NAME_OFFSET
+	multiply #2, NAME_LENGTH, 4 ; handles player name, rival name, backup name, trendy phrase
```

Comment:
`shiftleft #2, NAME_LENGTH, 2` — it's still a better alternative when using immediates...